app/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/PollingNewMessagesWithWorkItemJob.scala

```diff
@@ -123,41 +124,55 @@ class PollingNewMessagesWithWorkItemJob @Inject()
       }
   }
 
+
+  def process(messages: Seq[IEMessage], exciseNumber: String): Future[Boolean] = {
+
+    saveMovementSuccess.set(true)
+    messages.foldLeft(successful(true)) { case (acc, x) =>
+      acc.flatMap {
+        _ => saveToDbAndSendNotification(x, exciseNumber)
+      }.map( isSaved => {
+        if(!isSaved) saveMovementSuccess.set(false)
+        isSaved
+      })
+    }
+  }
   private def processMessages(
-                        exciseNumber: String,
-                        consumptionResponse: EISConsumptionResponse
-                      ): Future[Boolean] = {
-
-    messageParser.extractMessages(consumptionResponse.message)
-      .foldLeft(successful(true)) { case (acc, x) =>
-        acc.flatMap {
-          _ => saveToDbAndSendNotification(x, exciseNumber)
-        }
-      }
+                               exciseNumber: String,
+                               consumptionResponse: EISConsumptionResponse
+                             )(implicit hc: HeaderCarrier): Future[Boolean] = {
+
+    val messages = messageParser.extractMessages(consumptionResponse.message)
+    if(messages.isEmpty) return successful(true)
+
+    for {
+      _ <- process(messages, exciseNumber)
+      success = saveMovementSuccess.get()
+    } yield { success match {
+      case true => newMessageService.acknowledgeMessage(exciseNumber); true
+      case _ => false
+    }}
   }
 
   private def saveToDbAndSendNotification(
-    message: IEMessage,
-    exciseNumber: String
-  )(implicit ec: ExecutionContext): Future[Boolean] = {
+                                           message: IEMessage,
+                                           exciseNumber: String
+                                         )(implicit ec: ExecutionContext): Future[Boolean] = {
 
     movementService.updateMovement(message, exciseNumber).map {
-      movements =>
-        movements.map(m =>
-        notificationService.sendNotification(exciseNumber, m, message.messageIdentifier))
-          .sequence
-          .map { case o => o.forall(response =>
-            response match {
-              case _: SuccessPushNotificationResponse => true
-              case _ => false
-            }
-          )}
+      case Nil => Future(false)
+      case movements =>
+        movements.map { movement =>
+          notificationService.sendNotification(exciseNumber, movement, message.messageIdentifier)
+        }.sequence
```

Comment:
I know here we do not care if push-notification fails, but I wander if for coreteness we should return false if this happens.

Reply:
It would have a knock on effect on how you define things elsewhere I think? 

Currently false means it didn't save and so in process the atomic boolean will be set to false - so if we return false for Push notification fails as well then we'd be getting the atomic boolean setting to false in that scenario too, which we don't want?

Reply:
removed the atomic boolean